api/src/test/java/org/openmrs/api/AdministrationServiceTest.java

```diff
@@ -556,6 +560,208 @@ public void getGlobalProperty_shouldGetPropertyInCaseInsensitiveWay() {
 		assertEquals(orig, noprop);
 	}
 	
+	@Test
+	public void filterGlobalPropertiesByViewPermissions_shouldFilterGlobalPropertiesIfUserIsNotAllowedToViewSomeGlobalProperties() {
+		executeDataSet(ADMIN_INITIAL_DATA_XML);
+		
+		int expectedSize = adminService.getAllGlobalProperties().size();
+		// create a new test global property and add view privileges
+		GlobalProperty property = new GlobalProperty();
+		property.setProperty("test_property");
+		property.setPropertyValue("test_property_value");
+		property.setViewPrivilege(Context.getUserService().getPrivilege("Some Privilege For View Global Properties"));
+		adminService.saveGlobalProperty(property);
+		// assert new test global property is saved properly
+		List<GlobalProperty> properties = adminService.getAllGlobalProperties();
+		assertEquals(expectedSize + 1, properties.size());
+		
+		if (Context.isAuthenticated()) {
+			Context.logout();
+		}
+		
+		// authenticate new user to test view permissions
+		Context.authenticate("test_user", "test");
+		// have to add privilege in order to be able to call getAllGlobalProperties() method for new user
+		Context.addProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
+		
+		properties = adminService.getAllGlobalProperties();
+		int actualSize = properties.size();
+		
+		Context.removeProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
+		Context.logout();
+		
+		assertEquals(actualSize, expectedSize);
+		assertTrue(!properties.contains(property));
+	}
+	
+	/**
+	 * @see org.openmrs.api.AdministrationService#getGlobalProperty(java.lang.String)
+	 */
+	@Test
+	public void getGlobalProperty_shouldFailIfUserHasNoPrivileges() {
+		executeDataSet(ADMIN_INITIAL_DATA_XML);
+		GlobalProperty property = getGlobalPropertyWithViewPrivilege();
+		
+		User user = Context.getUserService().getUserByUsername("test_user");
+		assertNotNull(user);
+		// left this user without required privilege
+		Context.becomeUser(user.getSystemId());
```

Comment:
Is there any reason why you have used becomeUser and yet in the previous test you used a different technique of logout and login as a different user?

Reply:
@dkayiwa no reason, changed this to a more easier logout and login approach, thanks for the advise🙂